libs/providers/flagd-web/src/lib/flagd-web-provider.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  CallbackClient,
+  CallOptions,
+  Code,
+  ConnectError,
+  createCallbackClient,
+  createConnectTransport,
+  createPromiseClient,
+  PromiseClient,
+} from '@bufbuild/connect-web';
+import { PartialMessage, Struct } from '@bufbuild/protobuf';
+import {
+  ErrorCode,
+  EvaluationContext,
+  EventProvider,
+  FlagValue,
+  JsonValue,
+  Logger,
+  Provider,
+  ProviderEvents,
+  ResolutionDetails,
+  StandardResolutionReasons,
+} from '@openfeature/js-sdk';
+import { EventEmitter } from 'events';
+import { Service } from '../proto/ts/schema/v1/schema_connectweb';
+import {
+  EventStreamResponse,
+  ResolveBooleanRequest,
+  ResolveBooleanResponse,
+  ResolveFloatRequest,
+  ResolveFloatResponse,
+  ResolveObjectRequest,
+  ResolveObjectResponse,
+  ResolveStringRequest,
+  ResolveStringResponse,
+} from '../proto/ts/schema/v1/schema_pb';
+import { FlagdProviderOptions, getOptions } from './options';
+import { FlagdCache } from './flagd-cache';
+import { LocalStorageFlagdCache } from './local-storage-flagd-cache';
+
+export const ERROR_DISABLED = 'DISABLED';
+
+const EVENT_CONFIGURATION_CHANGE = 'configuration_change';
+const EVENT_PROVIDER_READY = 'provider_ready';
```

Comment:
Should these events live in the SDK?

Reply:
There are equivalents to these in the [experimental SDK](https://github.com/open-feature/js-sdk/pull/316/files), but these are specifically the ones flagd sends, so these are internal to the flagd implementation.